translators/make_list_gitlog.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+mkdir -p list
+base=$PWD/lxqt
+outdir=$base/../list
+
+cd $base
+ts=`find . -name *_*$1.ts`
+rm -vf $outdir/translators_*$1.gitlog
+
+for fp in $ts ; do
+  cd $base
+  dir=${fp%/*}
+  fn=${fp##*/}
+
+  lang=${fp#*_}
+  lang=${lang%.ts}
+  outp=$outdir/translators_$lang.gitlog
+  cd $dir
+  echo "gitlog: $fn"
+  echo -e "\nF:$fn" >> $outp
+  git log --use-mailmap --no-merges --pretty=format:"%H" $fn | xargs -n1 git log -1 --numstat --shortstat --pretty=format:"H:%H%ns:%s%nA:%aN %aE%na:%an %ae%nD:%ai" >> $outp
```

Comment:
Recently, git-log can't get the translator's name and email.
```
qps $ git log -1 --numstat --shortstat --pretty=format:"H:%H%ns:%s%nA:%aN %aE%na:%an %ae%nD:%ai" 89a914dde31aa4249e10d136d4943fc0ba9b9b45
H:89a914dde31aa4249e10d136d4943fc0ba9b9b45
s:Weblate commits (#255)
A:LXQtBot 72348767+LXQtBot@users.noreply.github.com
a:LXQtBot 72348767+LXQtBot@users.noreply.github.com
D:2020-10-30 15:24:34 +0800
3       3       src/translations/qps_ja.ts
 1 file changed, 3 insertions(+), 3 deletions(-)
qps $ 
```
I'll think of another way.
I see https://github.com/lxqt/lxqt-about/issues/97#issuecomment-749189577.

Reply:
Probably because they are "co-authored by" in default?

I see no automatic way to generate reports in json in weblate atm
https://docs.weblate.org/en/weblate-4.2.1/devel/reporting.html#translation-progress-reporting

And as I edited those "enter" shortkey translations I'm cited for many languages. Last 30 days:

`
[{"Portuguese": [["smarquespt@gmail.com", "S\u00e9rgio Marques", 2], ["hugokarvalho@hotmail.com", "Hugo Carvalho", 1615]]}, {"Chinese (Simplified)": [["chen@kaijie.org", "Kaijie Chen", 349]]}, {"Danish": [["standreas@riseup.net", "Standreas", 1], ["scootergrisen@gmail.com", "scootergrisen", 111]]}, {"Finnish": [["teemu.sirvio@fmi.fi", "Teemu Sirvi\u00f6", 181]]}, {"Italian": [["saverio.brancaccio@gmail.com", "Saverio Brancaccio", 5], ["standreas@riseup.net", "Standreas", 145]]}, {"Japanese": [["ito32bit+github@gmail.com", "Masamichi Ito", 222]]}, {"German": [["standreas@riseup.net", "Standreas", 5], ["atalanttore@googlemail.com", "Ettore Atalan", 44]]}, {"Russian": [["fatzer2@gmail.com", "Golubev Alexander", 1], ["astankov.dmitry@gmail.com", "Dmitry Astankov", 49]]}, {"Lithuanian": [["hazap@hotmail.com", "Moo", 36]]}, {"Swedish": [["standreas@riseup.net", "Standreas", 1]]}, {"Spanish": [["standreas@riseup.net", "Standreas", 1], ["jonathangarciamallen@gmail.com", "jogama", 4]]}, {"Portuguese (Brazil)": [["standreas@riseup.net", "Standreas", 1], ["eltonfabricio10@gmail.com", "EltonFF", 44]]}, {"Turkish": [["ayhanyalcinsoy@pisilinux.org", "Dr. Ayhan YAL\u00c7INSOY", 11], ["madanadam@gmail.com", "Mehmet Akif", 108]]}, {"Hungarian": [["standreas@riseup.net", "Standreas", 1], ["acszoltan111@gmail.com", "zoli111", 213]]}, {"Czech": [["standreas@riseup.net", "Standreas", 1], ["pavel.borecki@gmail.com", "p-bo", 145]]}, {"Greek": [["dglent@free.fr", "Dimitrios Glentadakis", 3]]}, {"Bulgarian": [["standreas@riseup.net", "Standreas", 2], ["bestpanda4ever@gmail.com", "Viliyan Yordanov", 21], ["mkondarev@gmail.com", "Mincho Kondarev", 89], ["mkondarev@yahoo.de", "Mincho Kondarev", 3204]]}, {"Basque": [["bipoza@gmail.com", "Bittor Poza", 84]]}, {"Hebrew": [["sh.yaron@gmail.com", "Yaron Shahrabani", 97]]}, {"Dutch": [["standreas@riseup.net", "Standreas", 1]]}, {"Chinese (Traditional)": [["chen@kaijie.org", "Kaijie Chen", 1], ["yan12125@gmail.com", "Chih-Hsuan Yen", 3]]}]`

Reply:
I don't know much about "co-authored by".
However, I think that "Squash and merge" is related.

I don't seem to have permission:
![Weblate-permission_screen-2020-12-23_13-05-38](https://user-images.githubusercontent.com/52656419/102958799-15635200-4521-11eb-8484-7a76b606bb50.jpg)

Reply:
Created a permission for you, not sure if it works.
If not you could have admin rights.

Reply:
PS: thinking about this new rule is about "downloading translation reports", probably it will not work for this.

Reply:
I don't have permission yet. But I don't need permission now.

If a command can get "Translation reports" without admin rights, I use it in the shell script.
Command: `curl` or `wlc` or `wget`...